src/esp/physics/bullet/BulletArticulatedObject.cpp

```diff
@@ -285,8 +289,6 @@ void BulletArticulatedObject::setRootState(const Magnum::Matrix4& state) {
   btAlignedObjectArray<btVector3> scratch_m;
   btMultiBody_->forwardKinematics(scratch_q, scratch_m);
   btMultiBody_->updateCollisionObjectWorldTransforms(scratch_q, scratch_m);
-  // sync visual shapes
-  updateNodes();
```

Comment:
We return object/link transforms from their SceneNodes. If we query a link transform after this setter but before the dynamics step then the queried transform will be outdated. However, I understand the value of deferring this update whenever possible. Maybe we want to either get those transforms directly from Bullet instead of our SceneGraph, or have a way to set state to dirty and update relevant nodes when the state queries are called.

Reply:
Yeah, this is a good point, we should just query those from bullet